packages/ui/src/composables/usePlacementAliases.ts

```diff
@@ -1,18 +1,37 @@
-export type PlacementPosition = 'top' | 'bottom' | 'left' | 'right'
+type YPlacementPosition = 'top' | 'bottom'
+type XPlacementPosition = 'left' | 'right'
+type PlacementPosition = XPlacementPosition | YPlacementPosition
+type PlacementPositionWithDefault = PlacementPosition | 'auto'
+type PlacementAlias = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right' | 'left-top' | 'right-top' | 'left-bottom' | 'right-bottom'
+
 export type PlacementAlignment = 'start' | 'end' | 'center'
-export type Placement = PlacementPosition | 'auto' | `${PlacementPosition}-${PlacementAlignment}`
-export type PlacementAlias = 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right'
+export type Placement = PlacementPositionWithDefault | `${PlacementPosition}-${PlacementAlignment}`
 export type PlacementWithAlias = Placement | PlacementAlias
 
-export const placementsPositions = ['top', 'bottom', 'left', 'right']
-  .reduce((acc, position) => [...acc, position, `${position}-start`, `${position}-end`, `${position}-center`], ['auto'] as string[])
-
-export const placementAliasesPositions = ['top', 'bottom']
-  .flatMap(position => [`${position}-left`, `${position}-right`, `left-${position}`, `right-${position}`])
-
-export const placementsPositionsWithAliases = [...placementsPositions, ...placementAliasesPositions]
-
-export const aliasToPlacement: Record<string, Placement> = {
+const yPlacementPosition: YPlacementPosition[] = ['top', 'bottom']
+const xPlacementPosition: XPlacementPosition[] = ['left', 'right']
```

Comment:
Can be named as `verticalPlacement`/`horizontalPlacement`. But it's for your discretion.